matRad_calcParticleDose.m

```diff
@@ -202,72 +235,125 @@
+                        
+                        if isfield(dij,'mLETDose') 
+                            tmpLET = zeros(size(currIx,1),1);
+                            tmpLET(currIx(:,:,c)) = matRad_interp1(depths,machine.data(energyIx).LET,radDepths(currIx(:,:,c),1,c));    
+                            totalLET = totalLET + tmpLET;
+                        end
+                    end
                     
+                    doseTmpContainer{mod(counter-1,numOfBixelsContainer)+1,1} = sparse(VdoseGrid(ix),1,totalDose,dij.doseGrid.numOfVoxels,1);
+                    if isfield(dij,'mLETDose') 
+                        letDoseTmpContainer{mod(counter-1,numOfBixelsContainer)+1,1} = sparse(VdoseGrid(ix),1,totalDose.*totalLET,dij.doseGrid.numOfVoxels,1);
+                    end                    
+                else
+                    % calculate particle dose for bixel k on ray j of beam i
+                    bixelDose = matRad_calcParticleDoseBixel(...
+                        currRadDepths, ...
+                        currRadialDist_sq(currIx), ...
+                        sigmaIni_sq, ...
+                        machine.data(energyIx));                 
+
+                    % dij sampling is exluded for particles until we investigated the influence of voxel sampling for particles
+                    %relDoseThreshold   =  0.02;   % sample dose values beyond the relative dose
+                    %Type               = 'dose';
+                    %[currIx,bixelDose] = matRad_DijSampling(currIx,bixelDose,radDepths(currIx),radialDist_sq(currIx),Type,relDoseThreshold);
+
+                    % Save dose for every bixel in cell array
+                    doseTmpContainer{mod(counter-1,numOfBixelsContainer)+1,1} = sparse(VdoseGrid(ix(currIx)),1,bixelDose,dij.doseGrid.numOfVoxels,1);
                 end
-                                
-                % calculate particle dose for bixel k on ray j of beam i
-                bixelDose = matRad_calcParticleDoseBixel(...
-                    currRadDepths, ...
-                    radialDist_sq(currIx), ...
-                    sigmaIni_sq, ...
-                    machine.data(energyIx));                 
-  
-                % dij sampling is exluded for particles until we investigated the influence of voxel sampling for particles
-                %relDoseThreshold   =  0.02;   % sample dose values beyond the relative dose
-                %Type               = 'dose';
-                %[currIx,bixelDose] = matRad_DijSampling(currIx,bixelDose,radDepths(currIx),radialDist_sq(currIx),Type,relDoseThreshold);
-                
-                % Save dose for every bixel in cell array
-                doseTmpContainer{mod(counter-1,numOfBixelsContainer)+1,1} = sparse(VdoseGrid(ix(currIx)),1,bixelDose,dij.doseGrid.numOfVoxels,1);
 
-                if isfield(dij,'mLETDose')
+                if isfield(dij,'mLETDose') && ~strcmp(pbCalcMode, 'fineSampling')
```

Comment:
Could we mabye move the LET computation within the else clause above? I guess that would be more consistent with how the LET is computed during fine sampling.